clang/lib/Sema/SemaType.cpp

```diff
@@ -7024,11 +7024,12 @@ static bool handleFunctionTypeAttr(TypeProcessingState &state, ParsedAttr &attr,
                << (int)Sema::CallingConventionIgnoredReason::VariadicFunction;
 
       attr.setInvalid();
-      if (S.getLangOpts().SYCLIsDevice) {
-        S.SYCLDiagIfDeviceCode(attr.getLoc(), diag::err_cconv_varargs)
-          << FunctionType::getNameForCallConv(CC);
-        return true;
-      } else
+      llvm::Triple Triple = S.Context.getTargetInfo().getTriple();
+      if (S.getLangOpts().SYCLIsDevice &&
+          S.DelayedDiagnostics.shouldDelayDiagnostics())
+        // No diagnostic e.g. on Microsoft system headers and SPIR compilation
+        // FIXME: further improvements needed here
```

Comment:
I had some wrong impression that this code would be not needed after the fix for deferred diagnostics for variadic arg functions (https://github.com/intel/llvm/pull/324). That is not quite correct as var-arg-funcs still fail on Windows.
If this code is still needed (i.e. it is the last/best), then please remove 'FIXME' comment 
or add more details to it explaining what exactly is wrong with this code and what needs to be done in future.